account-kit/smart-contracts/src/ma-v2/account/modularAccountV2.ts

```diff
@@ -182,8 +268,25 @@ export async function createModularAccountV2(
     }
   })();
 
-  return createMAv2Base({
-    source: "ModularAccountV2",
+  if (!signer) {
+    if (!credential)
+      throw new Error("Missing credential for MAV2 mode webauthn");
+    return await createMAv2Base({
+      source: "ModularAccountV2", // TO DO: remove need to pass in source?
```

Comment:
ideally the source is actually different by sub type cc @adam-alchemy @jakehobbs any thoughts here? will `source` make it easier in wallet-server logic?

Reply:
The only place we check `source` currently in wallet server is to check if it's `ModularAccountV2` to know that it supports permissions.

Not sure if that changes a bit in what @Zer0dot is working on for supporting other account types though.